cpa/tester/result.go

```diff
@@ -0,0 +1,185 @@
+package tester
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/CircleCI-Public/circle-policy-agent/cpa"
+	"github.com/CircleCI-Public/circle-policy-agent/internal"
+	"gopkg.in/yaml.v3"
+)
+
+type Result struct {
+	Group string
+	Name  string
+	Ok    bool
+	Err   error
```

Comment:
error is more like message, for example "no rules" isn't really an error.

Reply:
I wanted to change the type to string at some point, but then I can't easily test for Error types, ErrNoPolicies, ErrNoTests, or future known errors, and then make an informed decision about whether the test is passing or not.